miio/tests/test_protocol.py

```diff
@@ -32,3 +34,54 @@ def test_invalid_token(self):
             Utils.encrypt(payload, wrong_length)
         with self.assertRaises(ValueError):
             Utils.decrypt(payload, wrong_length)
+
+    def test_decode_json_payload(self):
+        token = bytes.fromhex(32 * '0')
+        ctx = {'token': token}
+
+        def build_msg(data):
+            encrypted_data = Utils.encrypt(data, token)
+
+            # header
+            magic = binascii.unhexlify(b'2131')
+            length = (32 + len(encrypted_data)).to_bytes(2, byteorder='big')
+            unknown = binascii.unhexlify(b'00000000')
+            did = binascii.unhexlify(b'01234567')
+            epoch = binascii.unhexlify(b'00000000')
+
+            checksum = Utils.md5(magic+length+unknown+did+epoch+token+encrypted_data)
+
+            return magic+length+unknown+did+epoch+checksum+encrypted_data
+
+        # can parse message with valid json
+        serialized_msg = build_msg(b'{"id": 123456}')
+
+        parsed_msg = Message.parse(serialized_msg, **ctx)
+        self.assertIsNotNone(parsed_msg)
+        self.assertIsNotNone(parsed_msg.data)
+        self.assertIsNotNone(parsed_msg.data.value)
+        self.assertIsInstance(parsed_msg.data.value, dict)
+        self.assertEqual(parsed_msg.data.value['id'], 123456)
```

Comment:
See the above comment about pytest & assert, also https://docs.pytest.org/en/latest/assert.html .